src/Serializer/ArticleVersionNormalizer.php

```diff
@@ -154,6 +166,14 @@ public static function articleClass(string $type, string $status = null)
         $data['versionDate'] = !empty($data['versionDate']) ? DateTimeImmutable::createFromFormat(DATE_ATOM, $data['versionDate']) : null;
         $data['statusDate'] = !empty($data['statusDate']) ? DateTimeImmutable::createFromFormat(DATE_ATOM, $data['statusDate']) : null;
 
+        $data['relatedArticles'] = $data['relatedArticles']
+            ->map(Closure::bind(function (array $article) use ($format, $context) {
+                // Get type.
+                $type = $this->articleClass($article['type'], $article['status'] ?? null);
+                // Denorm.
+                return $this->denormalizer->denormalize($article, $type, $format, $context);
```

Comment:
This is overly complicated, `$type` can just be `Article::class` so there's no need to work it out (like https://github.com/elifesciences/api-sdk-php/blob/master/src/Serializer/CollectionNormalizer.php#L73-L81).